src/node/stage/approved.rs

```diff
@@ -1322,305 +1499,125 @@ impl Approved {
-    }
+            .update_our_section(elders_info, section_key);
+        let neighbour_elders_removed = neighbour_elders_removed.build(&self.shared_state.sections);
+        self.prune_neighbour_connections(core, &neighbour_elders_removed);
 
-    fn add_sibling_elders_info(&mut self, elders_info: EldersInfo, section_key: bls::PublicKey) {
-        let prefix = elders_info.prefix;
-        self.shared_state.sections.add_neighbour(elders_info);
-        self.shared_state.sections.update_keys(prefix, section_key);
+        self.churn_in_progress = false;
 
-        // We can update their knowledge already because we know they also reached consensus on
-        // our `SectionInfo` so they know our latest key.
-        self.shared_state
-            .sections
-            .update_knowledge(prefix, self.shared_state.our_history.last_key_index());
+        Ok(())
     }
 
-    fn handle_neighbour_info_event(
-        &mut self,
-        core: &mut Core,
-        elders_info: EldersInfo,
-        key: bls::PublicKey,
-    ) -> Result<()> {
-        info!("handle NeighbourInfo: {:?}", elders_info);
-
+    fn update_neighbour_info(&mut self, core: &mut Core, elders_info: Proven<EldersInfo>) {
         let neighbour_elders_removed = NeighbourEldersRemoved::builder(&self.shared_state.sections);
-        self.shared_state
-            .sections
-            .update_keys(elders_info.prefix, key);
         self.shared_state.sections.add_neighbour(elders_info);
         let neighbour_elders_removed = neighbour_elders_removed.build(&self.shared_state.sections);
-
-        if !self.is_our_elder(core.id()) {
-            return Ok(());
-        }
-
         self.prune_neighbour_connections(core, &neighbour_elders_removed);
-        Ok(())
     }
 
-    fn handle_send_neighbour_info_event(
-        &mut self,
-        core: &mut Core,
-        dst: XorName,
-        nonce: MessageHash,
-    ) -> Result<()> {
-        if !self.is_our_elder(core.id()) {
-            return Ok(());
-        }
+    ////////////////////////////////////////////////////////////////////////////
+    // Parsec and Chain management
+    ////////////////////////////////////////////////////////////////////////////
 
-        self.send_routing_message(
-            core,
-            SrcLocation::Section(*self.shared_state.our_prefix()),
-            DstLocation::Section(dst),
-            Variant::NeighbourInfo {
-                elders_info: self.shared_state.our_info().clone(),
-                nonce,
-            },
-            None,
-        )
-    }
+    fn reset_parsec(&mut self, core: &mut Core, new_parsec_version: u64) -> Result<()> {
+        let is_elder = self.is_our_elder(core.id());
 
-    fn handle_their_key_info_event(&mut self, prefix: Prefix<XorName>, key: bls::PublicKey) {
-        self.shared_state.sections.update_keys(prefix, key);
-    }
+        let events = self.consensus_engine.prepare_reset(core.name());
+        let events = self.filter_events_to_revote(events);
```

Comment:
better not to use same variable name
why not just do:
```
let events = self.filter_events_to_revote(self.consensus_engine.prepare_reset(core.name()));
```

Reply:
Your suggestion fails the borrow checker